src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_global_properties_request.cc

```diff
@@ -820,6 +831,112 @@ bool SetGlobalPropertiesRequest::IsWhiteSpaceExist() {
+          custom_keys_array->end();
+
+      for (; it_keys != it_keys_end; ++it_keys) {
+        str = (*it_keys).asCharArray();
+        if (!CheckSyntax(str)) {
+          SDL_LOG_ERROR(
+              "Invalid keyboard_properties "
+              "customize_keys syntax check failed");
+          return true;
+        }
+      }
+    }
+  }
+
+  return false;
+}
+
+hmi_apis::Common_KeyboardLayout::eType
+SetGlobalPropertiesRequest::GetKeyboardLayout() {
+  SDL_LOG_AUTO_TRACE();
+
+  const smart_objects::SmartObject& msg_params =
+      (*message_)[strings::msg_params];
+  if (msg_params[strings::keyboard_properties].keyExists(
+          hmi_request::keyboard_layout)) {
+    return static_cast<hmi_apis::Common_KeyboardLayout::eType>(
+        msg_params[strings::keyboard_properties][hmi_request::keyboard_layout]
+            .asInt());
+  }
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+  auto saved_keyboard_props = app->keyboard_props();
+  if (saved_keyboard_props) {
+    if (saved_keyboard_props->keyExists(hmi_request::keyboard_layout)) {
+      return static_cast<hmi_apis::Common_KeyboardLayout::eType>(
+          (*saved_keyboard_props)[hmi_request::keyboard_layout].asInt());
+    }
+  }
+
+  return hmi_apis::Common_KeyboardLayout::QWERTY;
+}
+
+uint32_t SetGlobalPropertiesRequest::GetAllowedNumberOfConfigurableKeys() {
+  SDL_LOG_AUTO_TRACE();
+
+  auto display_capabilities = hmi_capabilities_.system_display_capabilities();
+  if (display_capabilities) {
+    auto* window_capabilities =
+        (*display_capabilities)[0][strings::window_capabilities].asArray();
+    if (window_capabilities) {
+      if ((*window_capabilities)[0].keyExists(
+              hmi_response::keyboard_capabilities)) {
+        if ((*window_capabilities)[0][hmi_response::keyboard_capabilities]
+                .keyExists(hmi_response::configurable_keys)) {
+          auto configurable_keyboards =
+              (*window_capabilities)[0][hmi_response::keyboard_capabilities]
+                                    [hmi_response::configurable_keys]
+                                        .asArray();
+
+          hmi_apis::Common_KeyboardLayout::eType requested_layout =
+              GetKeyboardLayout();
+          for (auto keyboard : (*configurable_keyboards)) {
+            if (requested_layout ==
+                static_cast<hmi_apis::Common_KeyboardLayout::eType>(
+                    keyboard[hmi_request::keyboard_layout].asInt())) {
+              return keyboard[hmi_response::num_configurable_keys].asUInt();
+            }
+          }
+        }
+      }
+    }
+  }
+  return 0u;
+}
+
+bool SetGlobalPropertiesRequest::IsCustomizableKeysDisallowed() {
+  SDL_LOG_AUTO_TRACE();
+
+  const smart_objects::SmartObject& msg_params =
+      (*message_)[strings::msg_params];
+
+  if (msg_params.keyExists(strings::keyboard_properties)) {
+    if (msg_params[strings::keyboard_properties].keyExists(
```

Comment:
@ychernysheva try the same approach as above

Reply:
Done in f10bc21